### ISSUE TITLE
Refactoring causing RecursionError

### ISSUE THREAD

**Issue by MarcoGorelli** [sourcery-ai/sourcery#96]
Hi,

I tried using Sourcery to refactor some of the pandas code, and it suggested a refactoring which caused a RecursionError:
```diff
-        for i in range(len(self)):
-            yield self[i]
+        yield from self
```
See https://github.com/pandas-dev/pandas/pull/37581

Flagging this up in case it's useful to you

**Comment by Hellebore** [comment]
Thanks for raising! We'll take a look.

**Comment by MarcoGorelli** [comment]
No worries! [here](https://github.com/pandas-dev/pandas/pull/37584#issuecomment-721893333)'s another refactoring which didn't work for us

**Comment by Hellebore** [comment]
> No worries! [here](https://github.com/pandas-dev/pandas/pull/37584#issuecomment-721893333)'s another refactoring which didn't work for us

That's an interesting one - we rely on the typing to infer that it's a list so we can do that refactoring.

**Comment by MarcoGorelli** [comment]
Ah OK, so in that one the refactoring brought up an incorrect type annotation...thanks!

**Comment by brendanator** [comment]
Thanks for the report. Fix identified and will be in the next release

**Comment by Hellebore** [comment]
Resolved with release of 0.8.3
